### PR TITLE
New Ruby Cheat Sheet

### PR DESCRIPTION
**Pull request by Sayanc93**
screen shot 2016-02-02 at 1 13 06 am

IA url: https://duck.co/ia/view/ruby_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@Sayanc93 Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/ruby.json [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+    "name": "Ruby",
+    "description": "Explanation of Ruby Methods v. 2.2.0",
+
+    "metadata": {
+        "sourceName": "RubyDoc",
+        "sourceUrl" : "http://ruby-doc.org/"
+    },
+
+    "aliases": [
+        "Ruby cheat sheet", "Cheat sheet for ruby", "ruby tutorial", "ruby syntax", "ruby help"
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/ruby.json]: We don't need `cheat sheet` in aliases. 

In-fact, remove all of these - these sorts of aliases should be dealt with by the `CheatSheets` Goodie, use aliases if there are different names and/or abbreviations for the subject.

**Comment by Sayanc93** [comment]
If anyone can help why the test is failing? @daxtheduck

File: share/goodie/cheat_sheets/json/ruby.json [hunk 2]
```diff
@@ -0,0 +1,245 @@
+{
+    "id": "ruby_cheat_sheet",
+    "name": "Ruby",
+    "description": "Explanation of Ruby Methods v. 2.2.0",
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/ruby.json]: Perhaps `Explanation of Ruby 2.2.0 Methods`?

**Comment by Sayanc93** [comment]
@GuiltyDolphin - Thanks for the suggestion. any idea why the test is failing?

**Comment by GuiltyDolphin** [comment]
@Sayanc93 It is a clash between the triggers and your aliases, try what I suggested and it should fix it.

**Comment by Sayanc93** [comment]
@GuiltyDolphin - The tests passed! Thanks for the help :)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Ruby cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/ruby_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Ruby cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/ruby_cheat_sheet_mobile.png)

**Comment by Sayanc93** [comment]
@GuiltyDolphin - Hey my tests are failing again. Any hints?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Ruby cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/ruby_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Ruby cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/ruby_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Ruby cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/ruby_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/ruby.json [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+                "key": "replace(other_string)",
+                "val": "Replaces the contents and taintedness of string with the corresponding values in other_string."
+            },
+            {
+                "key": "center(width, padstring=' ')",
+                "val": "Centers string in width. If width is greater than the length of string, returns a new string of length width with string centered and padded with padstring; otherwise, returns string."
+            },
+            {
+                "key": "reverse",
+                "val": "Returns a new string with the characters from string in reverse order."
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/ruby.json]: remove `.` at the end of val keys

**Comment by gautamkrishnar** [comment]
@moollaza why `count([other_string]+)` is working without using `count(\\[other_string\\]+)`?

**Comment by moollaza** [comment]
@gautamkrishnar if there are no spaces in the string, then the `[]{}` aren't used to separate keys. They only need to be escaped when there's spaces. Any string without spaces is taken literally

**Comment by Sayanc93** [comment]
@gautamkrishnar - removed periods at the end of vals. I hope this pr can be merged asap now? :)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Ruby cheat sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/ruby_cheat_sheet_mobile.png)

**Comment by Sayanc93** [comment]
@moollaza - hey i removed the periods at the end of vals but why do i still see them in the image generated by dax the duck? :/

File: share/goodie/cheat_sheets/json/ruby.json [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  "metadata": {
+    "sourceName": "RubyDoc",
+    "sourceUrl": "http://ruby-doc.org/"
+  },
+
+  "aliases": [
+    "ruby language", "rdoc"
+  ],
+
+  "template_type": "code",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/ruby.json]: I think we should use the `terminal` template to accommodate the lengthy descriptions

**Comment by moollaza** [comment]
@Sayanc93 apologies for the delay! This LGTM :+1:

**Pull request merged**